contracts/VRFConsumer.sol

```diff
@@ -1,77 +1,60 @@
-
-  error OnlyOwner(address notOwner);
-
-  modifier onlyOwner() {
-      if (msg.sender != sOwner) {
-          revert OnlyOwner(msg.sender);
-      }
-      _;
-  }
-
-  constructor(address coordinator) VRFConsumerBase(coordinator) {
-      sOwner = msg.sender;
-      COORDINATOR = IVRFCoordinator(coordinator);
-  }
-
-  // Receive remaining payment from requestRandomWordsPayment
-  receive() external payable {}
-
-  function requestRandomWords(
-      bytes32 keyHash,
-      uint64 accId,
-      uint32 callbackGasLimit,
-      uint32 numWords
-  )
-      public
-      onlyOwner
-      returns (uint256 requestId)
-  {
-    requestId = COORDINATOR.requestRandomWords(
-      keyHash,
-      accId,
-      callbackGasLimit,
-      numWords
-    );
-  }
-
-  function requestRandomWordsDirect(
-      bytes32 keyHash,
-      uint32 callbackGasLimit,
-      uint32 numWords
-  )
-      public
-      payable
-      onlyOwner
-      returns (uint256 requestId)
-  {
-    requestId = COORDINATOR.requestRandomWords{value: msg.value}(
-      keyHash,
-      callbackGasLimit,
-      numWords,
-      address(this)
-    );
-  }
-
-  function fulfillRandomWords(
-      uint256 /* requestId */,
-      uint256[] memory randomWords
-  )
-      internal
-      override
-  {
-    // requestId should be checked if it matches the expected request
-    // Generate random value between 1 and 50.
-    sRandomWord = (randomWords[0] % 50) + 1;
-  }
+    uint256 public sRandomWord;
+    address private sOwner;
+
+    IVRFCoordinator COORDINATOR;
+
+    error OnlyOwner(address notOwner);
+
+    modifier onlyOwner() {
+        if (msg.sender != sOwner) {
+            revert OnlyOwner(msg.sender);
+        }
+        _;
+    }
+
+    constructor(address coordinator) VRFConsumerBase(coordinator) {
+        sOwner = msg.sender;
+        COORDINATOR = IVRFCoordinator(coordinator);
+    }
+
+    // Receive remaining payment from requestRandomWordsPayment
+    receive() external payable {}
+
+    function requestRandomWords(
+        bytes32 keyHash,
+        uint64 accId,
+        uint32 callbackGasLimit,
+        uint32 numWords
+    ) public onlyOwner returns (uint256 requestId) {
+        requestId = COORDINATOR.requestRandomWords(keyHash, accId, callbackGasLimit, numWords);
+    }
+
+    function requestRandomWordsDirect(
+        bytes32 keyHash,
+        uint32 callbackGasLimit,
+        uint32 numWords
+    ) public payable onlyOwner returns (uint256 requestId) {
+        requestId = COORDINATOR.requestRandomWords{value: msg.value}(
+            keyHash,
+            callbackGasLimit,
+            numWords,
+            address(this)
+        );
+    }
+
+    function fulfillRandomWords(
+        uint256 /* requestId */,
+        uint256[] memory randomWords
+    ) internal override {
+        // requestId should be checked if it matches the expected request
+        // Generate random value between 1 and 50.
+        sRandomWord = (randomWords[0] % 50) + 1;
+    }
```

Comment:
@bayram98 Can you please also add a `withdrawTemporary` to this mock contract? Without that, consumer could not get money back if they request VRF through temporary account is not fulfilled.